src/app/queries/DepositHandler/checkDeposited.ts

```diff
@@ -0,0 +1,31 @@
+import { readContract } from "wagmi/actions";
+import { config } from "@/configs/wagmi";
+import { DepositHandlerAbi } from "@/abi/DepositHandler";
+import { Address } from "viem";
+import { Deposit } from "./DepositTypes";
+
+
+export async function checkDeposited(address: Address, bootcampAddress:Address) : Promise< Deposit | null | Error> {
+  try {
+    const result = await readContract(config, {
+      abi: DepositHandlerAbi,
+      address: bootcampAddress,
+      functionName: "deposits",
+      args: [
+        address
+      ],
+    });
+
+    if (result[0] !== 0n || result[1] !== false || result[2] !== 0) {
```

Comment:
the result var here can be destructured so it is more obvious what this if statement is checking at first glance

Reply:
It is a mistake, would raise an error if deposit is zero, so no need to check. I destructured indeed as it is way better that way.

Reply:
Fixed